tests/cpgrid/grid_lgr_test.cpp

```diff
@@ -79,116 +101,426 @@ void check_refinedPatch_grid(const std::array<int,3> cells_per_dim,
+                    }
+                    if (!touch_patch_onTopFace)
+                    {
+                        face_count +=1;
+                    }
+                    BOOST_CHECK( leaf_cell_to_face.size() == face_count);
+                } // end else
+            }
+        } // end-level-for-loop
+
+        BOOST_CHECK( static_cast<int>(startIJK_vec.size()) == coarse_grid.maxLevel());
+        BOOST_CHECK( (*data[data.size()-1]).parent_to_children_cells_.empty());
+
+        for (long unsigned int l = 0; l < startIJK_vec.size() +1; ++l) // level 0,1,2,... , last patch
+        {
+            const auto& view = coarse_grid.levelGridView(l);
+            for (const auto& element: elements(view)){
+                BOOST_CHECK_EQUAL(element.level(), l);
+            }
         }
-    }
 
-    const auto& leaf_view = coarse_grid.leafGridView();
-    for (const auto& element: elements(leaf_view)){
-         BOOST_CHECK( ((element.level() == 0) || (element.level() == 1)));
+        std::vector<int> leaf_to_parent_cell; // To store parent cell index, when leaf cell has a parent. Empty entry otherwise.
+        leaf_to_parent_cell.resize(data[startIJK_vec.size()+1]-> size(0)); // Correct size.
+        //
+        const auto& leaf_view = coarse_grid.leafGridView();
+        const auto& level0_view = coarse_grid.levelGridView(0);
+        Dune::MultipleCodimMultipleGeomTypeMapper<Dune::CpGrid::LeafGridView> leafMapper(leaf_view, Dune::mcmgElementLayout());
+        Dune::MultipleCodimMultipleGeomTypeMapper<Dune::CpGrid::LevelGridView> level0Mapper(level0_view, Dune::mcmgElementLayout());
+        const auto& leaf_idSet = (*data[startIJK_vec.size()+1]).local_id_set_;
+        const auto& level0_idSet = (*data[0]).local_id_set_;
+        for (const auto& element: elements(leaf_view)){
+            BOOST_CHECK( ((element.level() >= 0) || (element.level() < static_cast<int>(startIJK_vec.size()) +1)));
+            if (element.hasFather()) { // leaf_cell has a father!
+                leaf_to_parent_cell[leafMapper.index(element)] = level0Mapper.index(element.father());
+                const auto& id = (*leaf_idSet).id(element);
+                const auto& parent_id = (*level0_idSet).id(element.father());
+                std::cout << "Leaf cell: " <<  element.index() << ", element.index(): " <<  element.index() << " and id:" << id << '\n';
+                std::cout << "Father index: " << element.father().index() <<  " is equal to : "
+                          << leaf_to_parent_cell[element.index()] << " and parent_id: " << parent_id << '\n';
+                std::cout << "Level cell index: " << (*data[startIJK_vec.size()+1]).leaf_to_level_cells_[element.index()][1]
+                          << " in level: " << (*data[startIJK_vec.size()+1]).leaf_to_level_cells_[element.index()][0] << '\n';
+            }   
+        }
+        std::cout << '\n';
+        std::cout << "LeafMapper size: " << leafMapper.size() << '\n';
```

Comment:
Cool that you already started using it with the DUNE mappers. :+1: 
Is it intentional to have this as part of this PR or should we raher put into a separate one?
On the other hand it can't hurt as long as jenkins says yes. :thinking:

Reply:
I would keep it, if jenkins does not complain.